contracts/lib/math/SafeMath64.sol

```diff
@@ -14,17 +14,10 @@ library SafeMath64 {
     * @dev Multiplies two numbers, reverts on overflow.
     */
     function mul(uint64 _a, uint64 _b) internal pure returns (uint64) {
-        // Gas optimization: this is cheaper than requiring 'a' not being zero, but the
-        // benefit is lost if 'b' is also tested.
-        // See: https://github.com/OpenZeppelin/openzeppelin-solidity/pull/522
-        if (_a == 0) {
-            return 0;
-        }
+        uint256 c = uint256(_a) * uint256(_b);
+        require(c < 0x010000000000000000);
```

Comment:
Maybe we could use a `MAX_UINT64` constant here instead?

Reply:
I tried several options on Remix (like that one, `1 << 64`, `0xFFFFFFFFFFFFFFFF`, `2**64`) and this one seemed to the be the one using less gas.

Reply:
Oh weird, the constant should just get replaced by the compiler.

In that case, a comment would help :).

Reply:
Yes, that what I thought, but it seems not. I'll add a comment.